Bluetooth: Mesh: Update spec ref 1.0.1->1.1

Andrewpini opened the pull request:
Updates Bluetooth mesh specification references:

- Change "Profile" to "Protocol" since the main specification has changed its name.
- Update/align formating of specification references. This will make it easier to find spec references in the future.
- Change some section references to point to the correct section of the newest version of the specification (v1.1).

PavelVPV commented on doc/connectivity/bluetooth/api/mesh/access.rst:
```diff
@@ -171,9 +171,9 @@ Composition Data
 
    The implementation of the Bluetooth Mesh Protocol Specification version 1.1
-   is currently in an experimental state. For Bluetooth Mesh Profile Specification
+   is currently in an experimental state. For Bluetooth Mesh Protocol Specification
```
1.0.1 is still called `Profile`: https://www.bluetooth.com/specifications/specs/mesh-profile-1-0-1/ I guess we can't change this

PavelVPV commented on subsys/bluetooth/mesh/cfg_srv.c:
```diff
@@ -2305,7 +2305,7 @@ static int heartbeat_pub_set(struct bt_mesh_model *model,
 	pub.dst = sys_le16_to_cpu(param->dst);
 	if (param->count_log == 0x11) {
-		/* Special case defined in Mesh Profile Errata 11737 */
+		/* Special case defined in MshPRT-v1.1 Errata 11737 */
```
I'm uncertain about this. I guess the erratum was created to 1.0.1 spec, not to 1.1.

Andrewpini replied:
I found the same errata number (11737) in 1.1 spec as well. My thought was that it is easier to reference as little to the 1.0.1 as possible so that the reader mainly can reference the new spec.

PavelVPV replied:
It is mentioned in the MshPRTv1.1 but as I understand it was filed to 1.0.1. But now I read this as it was filed to 1.1 spec.

PavelVPV commented on subsys/bluetooth/mesh/transport_legacy.c:
```diff
@@ -870,11 +870,9 @@ static int trans_ack(struct bt_mesh_net_rx *rx, uint8_t hdr,
-		 * retransmit immediately when receiving a valid ack message.
-		 * Don't reset the retransmit timer if we didn't finish sending
-		 * segments.
+		/* According to MshPRT-v1.0.1: 3.5.3.3, we should reset the retransmit timer and
```
So here it is 1.0.1, not 1.1?

Andrewpini replied:
Since this is the legacy transport it made sence to reference 1.0.1. The legacy transport is for the 1.0.1 spec only right?

PavelVPV replied:
Ah, true, but then it should be MshPRFv1.0.1, not MshPRTv1.0.1

PavelVPV reviewed (commented):
@mia-ko , please have a look

PavelVPV reviewed (commented):
`-` should also probably be removed `MshPRT-v1.1` -> `MshPRTv1.1`.

PavelVPV commented on doc/connectivity/bluetooth/api/mesh/provisioning.rst:
```suggestion
insecure in the Mesh Profile 1.0.1 specification `erratum 16350 <https://www.bluetooth.org/docman/handlers/DownloadDoc.ashx?doc_id=516072>`_,
```

![image](https://github.com/zephyrproject-rtos/zephyr/assets/677018/3cdeebcb-a626-4a1f-9dbb-3bfb9757bf8d)

Andrewpini commented:
> `-` should also probably be removed `MshPRT-v1.1` -> `MshPRTv1.1`.

Sure thing 👍

Andrewpini committed:
Bluetooth: Mesh: Update spec ref 1.0.1->1.1

Updates Bluetooth mesh specification references:

- Change "Profile" to "Protocol" since the main specification has
changed its name.
- Update/align formating of specification references. This will
make it easier to find spec references in the future.
- Change some section references to point to the correct section of the
newest version of the specification (v1.1).

Signed-off-by: Anders Storrø <anders.storro@nordicsemi.no>

Balaklaka commented on subsys/bluetooth/mesh/access.c:
```diff
@@ -377,7 +377,7 @@ static int comp_add_elem(struct net_buf_simple *buf, struct bt_mesh_elem *elem,
 	if (net_buf_simple_tailroom(buf) < ((elem_size - *offset) + BT_MESH_MIC_SHORT)) {
 		if (IS_ENABLED(CONFIG_BT_MESH_LARGE_COMP_DATA_SRV)) {
-			/* Mesh Profile 1.1 Section 4.4.1.2.2:
+			/* MshPRTv1.1: 4.4.1.2.2:
```
Should there be a acronyme desription for MshPRTv1.1, MshPRFv1.0.1 and MshMBTv1.0, or linking to BT Sig with this. So the user knows what this squashed word means. There maybe someone that wonders what MshPRTv1.1, MshPRFv1.0.1 and MBTv1.0 stands for.

Andrewpini replied:
It is a fair point. If the majority of the references was directly in the documentation or API description I would definetly want a less cryptic acronym. When they are inside the stack implementation however I'm not sure if it is necessary. For most people that are qualified to go poking around in the stack source files I would expect that they att least can make a qualified guess what these acronyms refer to. But if you have a good suggestion where to place the acrony description I'm open to it 👍

Balaklaka reviewed (commented):
What about the .rst files? There is links to v1.0.1 in the documentation too. And possible in the Kconfig files. I know that for the CONFIG_BT_MESH_V1d1 it says that Mesh 1.1 is still in draft state.
I know that makes this tasks bigger, so it is also Ok to create a new task that looks at doc and kconfig.

Andrewpini commented:
> What about the .rst files? There is links to v1.0.1 in the documentation too. And possible in the Kconfig files. I know that for the CONFIG_BT_MESH_V1d1 it says that Mesh 1.1 is still in draft state. I know that makes this tasks bigger, so it is also Ok to create a new task that looks at doc and kconfig.

I've checked the Kconfig files and .rst, but I overlooked the root `mesh.rst` .  Im not sure that we should change `mesh.rst`  until we remove the experimental tag. I see that the v1.1 link is broken so I will fix that.

mia-ko commented on doc/connectivity/bluetooth/api/mesh/provisioning.rst:
```suggestion
insecure in the Bluetooth Mesh Profile Specification v1.0.1 `erratum 16350 <https://www.bluetooth.org/docman/handlers/DownloadDoc.ashx?doc_id=516072>`_,
```

mia-ko reviewed (changes requested):
Had just one small comment for one of the rst files.
Header files, you decide yourselves. I prefer it when the name of the spec is spelled out and not the short version like almost the file name. But I agree with you that whoever is reading this in comments of a header file, they would probably get it.

LingaoM commented:
<img width="573" alt="image" src="https://github.com/zephyrproject-rtos/zephyr/assets/26378606/ea6f26d5-f644-47bf-8533-ca3d01546854">

MshPRT define `provisionee`, so we should change all prov device to `provisionee`, also include rename prov_device.c to provisionee.c

Andrewpini commented:
> <img alt="image" width="573" src="https://user-images.githubusercontent.com/26378606/277204134-ea6f26d5-f644-47bf-8533-ca3d01546854.png">
> MshPRT define `provisionee`, so we should change all prov device to `provisionee`, also include rename prov_device.c to provisionee.c

I'll look in to it 👍

Andrewpini commented:
@LingaoM I have created a separate PR for the prov_dev->provisionee request: https://github.com/zephyrproject-rtos/zephyr/pull/64252

alxelax commented on subsys/bluetooth/mesh/cfg_srv.c:
```diff
@@ -2305,7 +2305,7 @@ static int heartbeat_pub_set(struct bt_mesh_model *model,
 	pub.dst = sys_le16_to_cpu(param->dst);
 	if (param->count_log == 0x11) {
-		/* Special case defined in Mesh Profile Errata 11737 */
+		/* Special case defined in MshPRFv1.0.1 Errata 11737 */
```
Isn't this considered already in Mesh v1.1?

alxelax reviewed (commented):
LGTM just a minor question.
